lib/mapper.js

```diff
@@ -314,6 +326,37 @@ function formatMetadata(props){
 }
 
 /**
+ * Flatten selectively based on your settings. You can either stringify, flatten, or drop the properties.
+ * Intercom rejects nested objects so you must choose a method.
+ *
+ * @param {Object} obj
+ * @param {Object} blacklisted
+ * @param {String} opts
+ * @return {Object} ret
+ * @api private
+ */
+
+function formatNested(obj, blacklisted, opts){
```

Comment:
maybe just have this take `obj, settings` and pull the map and default itself? that way we're not duping that logic in every method